ml-agents-envs/mlagents/envs/brain_conversion_utils.py

```diff
@@ -0,0 +1,66 @@
+from mlagents.envs.brain import BrainInfo, BrainParameters, CameraResolution
+from mlagents.envs.base_env import BatchedStepResult, AgentGroupSpec
+import numpy as np
+from typing import List, Any
+
+
+def step_result_to_brain_info(
+    step_result: BatchedStepResult,
+    group_spec: AgentGroupSpec,
+    agent_id_prefix: int = None,
+) -> BrainInfo:
+    n_agents = step_result.n_agents()
+    vis_obs_indices = []
+    vec_obs_indices = []
+    for index, observation in enumerate(step_result.obs):
+        if len(observation.shape) == 2:
+            vec_obs_indices.append(index)
+        if len(observation.shape) == 4:
+            vis_obs_indices.append(index)
```

Comment:
Can you raise here if the shape isn't 2 or 4?  This is all we ever send, but I'm worried about something getting silently dropped in the future.